src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -1699,7 +1696,7 @@ otError RadioSpinel<InterfaceType, ProcessContextType>::Remove(spinel_prop_key_t
 }
 
 template <typename InterfaceType, typename ProcessContextType>
-otError RadioSpinel<InterfaceType, ProcessContextType>::WaitResponse(void)
+otError RadioSpinel<InterfaceType, ProcessContextType>::WaitResponse(bool aWaitingReset)
```

Comment:
I think it's better to rename this as `aHandleRcpTimeout` to make this more general. 

And it's probably not necessary to add a method `WaitForResetReason`. Because I think `WaitForResetReason` is nothing different from `WaitResponse` except for if handling RCP timeout. The key factor that decides wait for what is the `mWaitingKey`. But `WaitForResetReason`has nothing to do with `mWaitingKey`. So maybe simply  call `WaitResponse` with different parameters in the code.